lib/model/Nutrient.dart

```diff
@@ -0,0 +1,309 @@
+    Nutrient.capricAcid: 'capric-acid',
+    Nutrient.lauricAcid: 'lauric-acid',
+    Nutrient.myristicAcid: 'myristic-acid',
+    Nutrient.palmiticAcid: 'palmitic-acid',
+    Nutrient.stearicAcid: 'stearic-acid',
+    Nutrient.oleicAcid: 'oleic-acid',
+    Nutrient.linoleicAcid: 'linoleic-acid',
+    Nutrient.docosahexaenoicAcid: 'docosahexaenoic-acid',
+    Nutrient.eicosapentaenoicAcid: 'eicosapentaenoic-acid',
+    Nutrient.erucicAcid: 'erucic-acid',
+    Nutrient.monounsaturatedFat: 'monounsaturated',
+    Nutrient.polyunsaturatedFat: 'polyunsaturated',
+    Nutrient.pantothenicAcid: 'pantothenic-acid',
+    Nutrient.omega3: 'omega-3-fat',
+    Nutrient.omega6: 'omega-6-fat',
+    Nutrient.transFat: 'trans-fat',
+  };
+
+  /// Default units for nutrients.
+  static const Map<Unit, Set<Nutrient>> _defaultUnits = <Unit, Set<Nutrient>>{
+    Unit.KCAL: <Nutrient>{
+      Nutrient.energyKCal,
+    },
+    Unit.KJ: <Nutrient>{
+      Nutrient.energyKJ,
+    },
+    Unit.PERCENT: <Nutrient>{
+      Nutrient.alcohol,
+    },
+    Unit.MILLI_G: <Nutrient>{
+      Nutrient.calcium,
+      Nutrient.iron,
+      Nutrient.vitaminC,
+      Nutrient.magnesium,
+      Nutrient.phosphorus,
+      Nutrient.potassium,
+      Nutrient.zinc,
+      Nutrient.copper,
+      Nutrient.vitaminE,
+      Nutrient.vitaminB1,
+      Nutrient.vitaminB2,
+      Nutrient.vitaminPP,
+      Nutrient.vitaminB6,
+      Nutrient.cholesterol,
+      Nutrient.pantothenicAcid,
+      Nutrient.chloride,
+      Nutrient.fluoride,
+      Nutrient.manganese,
+      Nutrient.omega3,
+      Nutrient.omega6,
+    },
+    Unit.G: <Nutrient>{
+      Nutrient.salt,
+      Nutrient.sodium,
+      Nutrient.fiber,
+      Nutrient.sugars,
+      Nutrient.fat,
+      Nutrient.saturatedFat,
+      Nutrient.proteins,
+      Nutrient.carbohydrates,
+      Nutrient.caffeine,
+      Nutrient.butyricAcid,
+      Nutrient.caproicAcid,
+      Nutrient.caprylicAcid,
+      Nutrient.capricAcid,
+      Nutrient.lauricAcid,
+      Nutrient.myristicAcid,
+      Nutrient.palmiticAcid,
+      Nutrient.stearicAcid,
+      Nutrient.oleicAcid,
+      Nutrient.linoleicAcid,
+      Nutrient.docosahexaenoicAcid,
+      Nutrient.eicosapentaenoicAcid,
+      Nutrient.erucicAcid,
+      Nutrient.monounsaturatedFat,
+      Nutrient.polyunsaturatedFat,
+      Nutrient.transFat,
+    },
+    Unit.MICRO_G: <Nutrient>{
+      Nutrient.selenium,
+      Nutrient.vitaminA,
+      Nutrient.vitaminD,
+      Nutrient.vitaminB12,
+      Nutrient.vitaminB9,
+      Nutrient.vitaminK,
+      Nutrient.biotin,
+      Nutrient.chromium,
+      Nutrient.iodine,
+      Nutrient.molybdenum,
+    },
+  };
+
+  /// Returns the tag used in [Nutriments]'s map.
+  String get tag =>
+      _specialNutrientTags[this] ?? toString().replaceFirst('Nutrient.', '');
+
+  /// Returns the default unit of values.
+  Unit? get defaultUnit {
+    for (final MapEntry<Unit, Set<Nutrient>> entry in _defaultUnits.entries) {
```

Comment:
This works, but why not use a simple map of nutrients to units so that it can be retrieved directly? (and that we can have the map defined in alphabetic order, to avoid bugs like having the same nutrient defined in 2 groups).